cmd/watch-termination/main.go

```diff
@@ -126,12 +127,31 @@ func run() int {
 			// keep going
 		}
 
-		defer func() {
+		var deleteLockOnce sync.Once
+
+		if *gracefulTerminatioPeriod > 2*time.Second {
+			go func() {
+				<-time.After(*gracefulTerminatioPeriod - 2*time.Second)
+
+				deleteLockOnce.Do(func() {
```

Comment:
calling an anonymous function is guaranteed to be safe? how does it know it is the same instance of the function?
the doc doesn't say a word on that.

Reply:
it does not matter it's the same. The state is in the `deleteLockOnce` object.

Reply:
```
// Do calls the function f if and only if Do is being called for the
// first time for this instance of Once. In other words, given
// 	var once Once
// if once.Do(f) is called multiple times, only the first call will invoke f,
// even if f has a different value in each invocation. A new instance of
// Once is required for each function to execute.
```

Reply:
ok, you are right - https://play.golang.org/p/xZ-LrKL325Q